### DISCUSSION TITLE
I want to contribute

### DISCUSSION THREAD

**Discussion by kamyar1979** [icerpc/icerpc-csharp#3945] [Ideas]
I can help you write other languages. The following languages are my skills:

C#
Python
Kotlin
Rust
D
C++
Lua

**Comment by bernardnormier** [comment]
Hi Kamyar,

Thank you, we certainly welcome contributions!

I want to write a blog outlining the steps to implement IceRPC in a new language on your own. It's a significant effort but definitely doable: IceRPC is fairly small. `icerpc-csharp/src` has only about 18,000 lines of C# code. And icerpc-csharp includes many features (ice protocol support, Slice1 support, DI support...) that you may not need/want in IceRPC for _Xxx_, especially in your initial version.

Now, we are not stopping with C# - we are also implementing IceRPC for Rust, and we will make the repo public as soon as soon as it's in a usable state.  Then I am sure there will be numerous tasks / issues to resolve before the initial release.

In the meantime, if you like, you could look at issues in icerpc-csharp--start with a small one!

Thanks,
Bernard
